File: src/my_project/easy_problems/from101to150/next_greater_element_1.py
from typing import List 

class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        flag: bool = False
        
        for i in range(len(nums1)):
            flag = False

            for j in range(len(nums2)):

                if nums1[i] == nums2[j]:
                    print(i,j)

                    for l in range(j+1, len(nums2)):
                        print("flag")
                        if nums2[l] > nums1[i]:
                            nums1[i] = nums2[l]
                            flag = True
                            break 
                    if flag == False:
                        nums1[i] = -1

                    break
                    
        return nums1

solution = Solution()

print(solution.nextGreaterElement(nums1 = [4,1,2], nums2 = [1,3,4,2]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/nextGreaterElement1.ts
function nextGreaterElement(nums1: number[], nums2: number[]): number[] {
    let flag: boolean = false

    for (let i = 0; i < nums1.length; i++){
        flag = false 
        for (let j = 0; j < nums2.length; j++){
            if (nums1[i] === nums2[j]){
                for (let l = j+1; l < nums2.length; l++){
                    if (nums2[l] > nums1[i]){
                        nums1[i] = nums2[l]
                        flag = true
                        break
                    }
                }
                if (flag === false){
                    nums1[i] = -1
                }
                break
            }
        }
    }
    return nums1
};
// new function to be reverted
function newFunctionToBeReverted(numVariable: number){
    return numVariable
}


console.log(nextGreaterElement([4,1,2], [1,3,4,2]))